cEP-0020.md

```diff
@@ -0,0 +1,244 @@
+gamification system which will allow us to track newcomer's progress and
+assign some points to each of the activity they do. There will be different levels
+based on points threshold and a newcomer will also get some special privileges
+after the compeletion of each level. There will also be different badges on the
+basis of performed activities by the newcomers.
+
+So, every newcomer at coala will have to complete the processes involved in
+the gamification system in order to become a developer at coala or to get the
+**coala Developer** badge.
+
+## Background
+
+### Newcomer Experience
+
+**The problems we have:**
+
+- How to attract more newcomers
+  - How to cope with the increasing amounts of newcomers
+- How to teach and help newcomers to learn faster so
+  we can keep working on coala at a high pace.
+- How to improve newcomer -> dev/maintainer ratio
+- How to increase amount of feedback we get
+  - How to use feedback to improve coala and processes
+
+## Why Gamification?
+
+As coala tries to be a welcoming organization for newcomers, it gives them
+clear pathways and as much direct assistance as possible. However, it is
+already known that newcomers face many barriers while attempting to contribute
+to open source for the first time, and additional barriers while attempting
+to contribute to projects run by the coala organisation.
+
+Some barriers they face include orientation issues that can potentially
+demotivate newcomers from placing their first contribution. On the other
+hand, gamification is widely used to engage and motivate people to
+accomplish tasks and improve their performance.
+
+Therefore, the goal of this project is to use gamification to orient and
+motivate newcomers to overcome onboarding barriers to contribute and engage
+with coala and its community.
+
+## Gamification Elements
+
+There will be four game design elements to help newcomers: Activities, Points,
+Levels and Badges. For each of these elements, we will have a set of rules to
+describe the operation of the gamified environment. i.e.:
+
+### Activities
+
+Let's start by defining some activities which could be tracked:
+
+- Create Issues
+  - Labels?
+  - No. of comments?
+- Create Pull Requests
+  - Label?
+  - CI status?
+  - No. of Gitmate errors?
+  - No. and type of Comments?
+- Comments
+  - Type?
+- Gitter Messages
+  - Good?
+  - Bad?
+
+The above activities will be tracked seperately based on their category. i.e.:
+
+- There are five types of Issues and Pull Requests that could be tracked:
+  - type/bug
+  - difficulty/newcomer
+  - difficulty/low
+  - difficulty/medium
+  - difficulty/high
+- While tracking Pull Requests, following factors will also be considered:
+  - CI status
+    - Passed
+    - Failed
+  - Comments
+    - No. of simple comments
+    - No. of review comments
+  - Gitmate Errors:
+    - 1 to 5
+    - 5 to 10
+    - 10 and beyond
+- Following are the types of comments that could be tracked:
+  - Issue comments
```

Comment:
'any comment' doesnt feel like an activity we want to award points to, as then we need lots of negatives to prevent bad comments.

What types of issue comments do we want, *and* can accurately without meta-review?

However 'Comment on an issue' can be an activity without any points.  It could be quite useful to clearly show which activities we give 0 points for, so people dont waste their time doing those types of activities.

In fact, (im not so sure about this) we may want to give all comments a preset negative points by default, because any good comment will also get a positive count somewhere else.